### PR TITLE
[test stats] use published test stats for sharding

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81166
* #81165
* #81164
* #81163
* __->__ #81116

Use the nightly-published test stats to perform sharding, instead of
calculating it in every build job.

### CODE REVIEW

File: tools/testing/test_selections.py [hunk 1]
```diff
@@ -58,37 +58,43 @@ def _calculate_job_times(reports: List["Report"]) -> Dict[str, float]:
 
 
 def calculate_shards(
-    num_shards: int, tests: List[str], job_times: Dict[str, float]
+    num_shards: int, test_files: List[str], test_file_times: Dict[str, float]
```
Comment (suo) [inline tools/testing/test_selections.py:61]: I didn't change any logic in this function; just renamed the variables to more closely match what we're doing.

**Review by janeyx99 — APPROVED** [review]
Code looks good, but could you verify that the same tests were run as before in the stats before landing?

Brief glance at the logs also looks good.

**Comment by suo** [comment]
image
num tests run looks fine

**Comment by suo** [comment]
Wait but the test jsons preserved at build time are downloaded to the test
jobs…from s3 lol

**Pull request closed without merging**